### ISSUE TITLE
[BUG] FreshRSS only saves last article of feed

### ISSUE THREAD

**Issue by mrnoname1000** [FreshRSS/FreshRSS#5329]
**Describe the bug**
FreshRSS only saves the last article of [this feed](https://megatools.megous.com/rss.xml). I suspect it's because the GUIDs only differ in the URL fragment, or everything after the `#`. Using XML+XPath scraping works around the issue.

**To Reproduce**
1. Add https://megatools.megous.com/rss.xml to FreshRSS
2. The only entry shown is the last one

**Expected behavior**
All entries are saved by FreshRSS

**Environment information (please complete the following information):**
- Device: ThinkPad A485
- OS: Void Linux
- Browser: Firefox 112
- FreshRSS version: edge [11572462]
- Database version: PostgreSQL 13.9
- PHP version: PHP 7.4
- Installation type: Self-hosted [Debian 11.6]

**Comment by mrnoname1000** [comment]
Oh, it's because all the articles have the same link. Using the GUID as the link is what made XML+XPath work.

**Comment by Alkarex** [comment]
Right, this feed has duplicated GUIDs (e.g. `http://megatools.megous.com/#2d538417e95484444cc66394a9e6e65b82b37922` which is there twice), so FreshRSS reverts to using the link as article identifier, which also happens to be the same for all articles...
We have an issue / discussion somewhere about allowing a manual override of which unique identifier to use for buggy feeds (PR welcome).
